aiida/cmdline/commands/cmd_group.py

```diff
@@ -84,6 +84,67 @@ def group_remove_nodes(group, nodes, clear, force):
         group.remove_nodes(nodes)
 
 
+@verdi_group.command('move-nodes')
+@arguments.NODES()
+@click.option('-s', '--source-group', type=types.GroupParamType(), required=True, help='The group whose nodes to move.')
+@click.option(
+    '-t', '--target-group', type=types.GroupParamType(), required=True, help='The group to which the nodes are moved.'
+)
+@options.FORCE(help='Do not ask for confirmation and skip all checks.')
+@options.ALL(help='Move all nodes from the source to the target group.')
+@with_dbenv()
+def group_move_nodes(source_group, target_group, force, nodes, all_entries):
+    """Move the specified NODES from one group to another."""
+    from aiida.orm import Group, Node, QueryBuilder
+
+    if not force and source_group.pk == target_group.pk:
+        echo.echo_critical(f'Source and target group are the same: {source_group}.')
```

Comment:
I wonder if skipping this check ever makes sense, even if `force` is used. It would be a no-op essentially wouldn't it? Can't see this ever making sense so best just raise.